include/bluetooth/audio/csis.h

```diff
@@ -327,21 +329,40 @@ bool bt_csis_client_is_set_member(uint8_t set_sirk[BT_CSIS_SET_SIRK_SIZE],
 void bt_csis_client_register_cb(struct bt_csis_client_cb *cb);
 
 /**
- * @brief Check if an array of set members are unlocked
+ * @brief Callback function definition for bt_csis_client_ordered_access()
  *
- * This will read the set lock value on all members and respond with a single
- * state.
+ * @param set_info   Pointer to the a specific set_info struct.
+ * @param members    Array of members ordered by rank. The procedure shall be
+ *                   done on the members in ascending order.
+ * @param count      Number of members in @p members.
  *
- * @param members   Array of set members to check lock state for.
+ * @return true if the procedures can be successfully done, or false to stop the
+ *         procedure.
+ */
+typedef bool (*bt_csis_client_ordered_access_t)(const struct bt_csis_client_set_info *set_info,
+						struct bt_csis_client_set_member *members[],
+						size_t count);
```

Comment:
I agree with all your points. 

> First of all, the bt_csis_client_ordered_access says that cb is called in the order of the member's ranks (if present) but the the callback does not have a member parameter, but the members (ordered by rank I suppose).
Thus, shouldn't this callback look like this ?

After a discussion with Asbjørn I changed it from a foreach to a single callback with all members. That seemed easier to use for all parties. 

> Secondly, you add another callback function named bt_csis_client_ordered_access_cb_t. That is confusing having those two with so similar names.

> Thirdly, Is it really needed to have 2 callbacks for this procedure? I have a feeling that the one given to bt_csis_client_ordered_access with additional err parameter (to indicate that procedure cannot be performed because one of the peer's is locked) should be sufficient.

One callback is called when the procedure is ready to be done on each member, and the other is when the ordered access procedure is completed. 

Ideally I'd like to reduce this to a single callback, but the parameters in each callback are sufficiently different that it doesn't really make sense IMO. 

I do, however, want to find a better name for these, as `_cb_` is the only difference between them... 

Basically we have
```c
bt_csis_client_ordered_access()
bt_csis_client_ordered_access_t() // callback
bt_csis_client_ordered_access_cb_t() // callback for bt_csis_client_ordered_access
```

Personally I think a new name for `bt_csis_client_ordered_access_t` is the best solution here. 

How do you feel about `bt_csis_client_ordered_access_ready_t` or `bt_csis_client_ordered_access_execute_t`? 
Do you have any other ideas, or perhaps an idea to reduce the callbacks to a single callback?

Reply:
> After a discussion with Asbjørn I changed it from a foreach to a single callback with all members. That seemed easier to use for all parties.

OK, but the documentation was out of date I guess. Thanks for update ^^ Maybe it would be nice to have a info that `@ref bt_csis_client_ordered_access_cb_t` will be called when procedure is completed.

Reply:
Added that in the documentation now as well :)